common/shareddb.h

```diff
@@ -119,25 +119,33 @@ class SharedDatabase : public Database {
 	 */
 	EQ::ItemInstance *CreateItem(
 		uint32 item_id,
-		int16 charges = 0,
-		uint32 aug1 = 0,
-		uint32 aug2 = 0,
-		uint32 aug3 = 0,
-		uint32 aug4 = 0,
-		uint32 aug5 = 0,
-		uint32 aug6 = 0,
-		bool attuned = 0
+		int16 charges,
+		uint32 aug1,
+		uint32 aug2,
+		uint32 aug3,
+		uint32 aug4,
+		uint32 aug5,
+		uint32 aug6,
+		bool attuned,
+		const std::string& custom_data,
```

Comment:
Why should these be required for every single item instance creation? This is very verbose to call `CreateItem`

Similarly, why did all the previous defaults get removed? It unnecessarily makes our existing 57 calls to `CreateItem` needlessly verbose.

I would think we could do 

```cpp
inst->SetItemCustomData();
inst->SetOrnamentIcon();
```

For optionally filled fields after instance creation

This feels like maybe you did this on accident

Reply:
I explained it in the post.

The database.CreateItem functions are more explicit in their usage, this adds a bit more boilerplate code in a few spots but it helps with seeing what's actually going on and where we could be losing data.

Reply:
I could undo it if it's an issue, but taking it out did lead me to discover a similar issue with guild banks which I will fix after this patch goes in either way.

Reply:
I read the PR description but doesn't necessarily register with this kind of change in my mind as I read it - but you did call it out.

```
adds a bit more boilerplate code in a few spots
```

If was maybe 2,3 spots sure, maybe we can just shrug it off - but this is a very commonly used function in the code called 53 times in this branch. It also adds 12 optional args as now required args in most of those 53 places. Typing that out in normal code writing would feel cumbersome in any circumstance.

![image](https://user-images.githubusercontent.com/3319450/226093426-e4603cb9-4800-428b-b936-9a0718077fdf.png)

```
helps with seeing what's actually going on and where we could be losing data
```

How does this help though? Outside of seeing everything in a quick search as args in this one instance of auditing this problem in the code. 

I agree with explicitness 100%, but arguably the optional parameters are a code smell and I'm amazed we've loaded this constructor function as much as we have as it is. https://blog.devgenius.io/code-smell-19-optional-arguments-c0714855dbbb

At the cost of extreme code verbosity it doesn't help readability in anyway. I'm having a hard time seeing these fields as required in most cases to be part of `CreateItem` when they can be set on the instance after the item is created in the places in the code where we need to mutate it (and any other item related thing we need to set)

We can trace all the places we call `CreateItem` and see if we do anything in that part of the code to mutate item related data after the instance has been instantiated

Outside of being able to see all calls to CreateItem and see where we're not passing some data in I don't really see the benefits outweighing the verbosity, readability, maintainability smells. It's just as easy to click through those parts of the code and see if we're calling the setters/getters to load/fetch those fields

I really do think these changes are great though, I don't want my strong thoughts on this signature change to invalidate that. I'm excited to see how this helps the rest of the GUID and instance work

Reply:
I've definitely come to appreciate being explicit with your intentions especially with code that is prime for misuse like that.  That said, I'm not married to the idea and will go through and change it back tomorrow.

Reply:
Thank you <3